jax/experimental/jax2tf/README.md

```diff
@@ -76,12 +81,62 @@ my_model.f(tf.ones([16, 28, 28]))  # a batch size of 16
 tf.saved_model.save(my_model, '/some/directory')
 ```
 
-More involved examples of using SavedModel, including how to prepare
-Flax models for conversion and saving, are described in the
+### Usage: saved models from Flax
```

Comment:
Shouldn't this section be in the Haiku or Flax documentation instead of in JAX?

Reply:
Perhaps it should. Maybe it will migrate there. For now, I feel that it makes sense for it to be hosted here, because from a Flax/Haiku perspective jax2tf is such a niche, and so fluid, and more intertwined with the way the `convert_and_save_model` works than with the libraries themselves.

Reply:
Is what I wrote about how to split a Haiku model into a predict function and a pytree of parameters correct?

Reply:
This touches on some organization and API aspects that I am still trying to optimize. On one hand, I think it would be strategically important for jax2tf to position itself as completely independent of SavedModel: you get a TF function and you package it as a SavedModel entirely as you wish. This is important because it addresses a lot of the worries of the form "will JAX SavedModel support X kind of assets or input_signatures or metadata"? Hence, my initial thinking was to NOT have an API that creates SavedModel. 

However, when constructing examples to work with Keras, TF Hub, TF.js, and Servo, in combination with Flax, it took quite a bit of time to figure out how to put everything together. It was in this process that I wrote the saved_model_lib.py, to be used in the examples. Last week I had to send twice over email the information for how to package Flax models for use with SavedModel. I think that I owe to our users to write down how this can be done. And I do think that *at the moment* the `convert_and_saved_model` documentation and how it is used with Flax and Haiku is too temporary to deserve being in the Flax or Haiku documentation. 

This through this comment makes me think to not include saved_model_lib.py as part of jax2tf library, but put it in jax2tf/examples. The documentation of how to use it with Flax and Haiku also goes in the examples for now. Eventually, we can migrate the documentation to Flax and Haiku. I would also prefer to not even export the library in google3, and tell people to just copy the code.

Reply:
It makes sense while you are iterating on this feature and it is alpha for the documentation to be in one place, I think once it is stable that we should move the documentation out of JAX and into the respective libraries (we can always x-link from JAX if that is useful).

Perhaps longer term the JAX+TF utility layer (e.g. **not** jax2tf, but things built on top of that) could be a separate library that can be evolved using the public API.